scripts/validation-spike.py

```diff
@@ -0,0 +1,311 @@
+    :return: Dict of project_id's mapped to project names for this ingest. .
+    """
+    fund_type = df_place.loc[
+        df_place["Question"] == "Are you filling this in for a Town Deal or Future High Street Fund?"
+    ]["Answer"].values[0]
+    place_name = df_place.loc[df_place["Question"] == "Please select your place name"]["Answer"].values[0]
+
+    # fetch either "Town Deal" or "Future High Streets Fund" project_id lookup table
+    df_lookup = df_lookup.iloc[2:, 1:4] if fund_type == "Town_Deal" else df_lookup.iloc[2:295, 8:11]
+    # hard-code column headers rather than extract from spreadsheet headers due to typo's in the latter.
+    df_lookup.columns = ["Unique Project Identifier", "Town", "Project Name"]
+
+    # filter on current place / programme and convert to dict
+    df_lookup = df_lookup.loc[df_lookup["Town"].str.lower().str.strip() == str(place_name).lower().strip()]
+    project_lookup = dict(zip(df_lookup["Project Name"], df_lookup["Unique Project Identifier"]))
+
+    return project_lookup
+
+
+# SPIKE covers the Risk Register sheet of the TF Round 3/4 template
+
+wb = pd.read_excel("validation-spike-template.xlsx", sheet_name=None)
+
+schema = {
+    "Programme Risk": {
+        "sheet": "7 - Risk Register",
+        "ID": "IDABC1",  # add this ID above the top left cell of any copies of this table in the template
+        "pandera_schema": pa.DataFrameSchema(
+            {
+                "Risk Name": pa.Column(str, unique=True, report_duplicates="exclude_first"),
+                "Risk Category": pa.Column(str, pa.Check.isin(enum.RiskCategoryEnum)),
+                "Short description of the Risk": pa.Column(int),
+                "Full Description": pa.Column(str),
+                "Consequences": pa.Column(str),
+                "Pre-mitigated Impact": pa.Column(str),
+                "Pre-mitigated Likelihood": pa.Column(str),
+                "Mitigations": pa.Column(str),
+                "Post-Mitigated Impact": pa.Column(str),
+                "Post-mitigated Likelihood": pa.Column(str),
+                "Proximity": pa.Column(str, pa.Check.isin(enum.ProximityEnum)),
+                "Risk Owner/Role": pa.Column(str),
+            },
+            index=pa.Index(int),
+            strict=True,
+            coerce=True,
+        ),
+        "schema": {
+            "rows_to_drop": [0],
+            "column_indexes": {
+                "Risk Name": "C",
+                "Risk Category": "D",
+                "Short description of the Risk": "E",
+                "Full Description": "F",
+                "Consequences": "G",
+                "Pre-mitigated Impact": "H",
+                "Pre-mitigated Likelihood": "I",
+                "Mitigations": "K",
+                "Post-Mitigated Impact": "L",
+                "Post-mitigated Likelihood": "M",
+                "Proximity": "N",
+                "Risk Owner/Role": "O",
+            },
+        },
+    },
```

Comment:
Probably requires some kind comment explaining the distinction between these two schemas.

Reply:
Hopefully the docstrings in the `TableSchema` initialiser help with this. Can also add more docs when it's properly written into the code base